run/start-all.mjs

```diff
@@ -60,11 +60,12 @@ async function main() {
     gatewayServers[serverName] = fork('../services/main/gateway-server.mjs', ['--service-name', serverName], { cwd: '.' });
   }
 
+  await new Promise(r => setTimeout(r, 4000)); // TODO: better way to prevent race condition.
```

Comment:
By the way, I think I need some help on this part. How do I ensure that standalone extension are executed **after** gateway services are fully started?
In some cases, a standalone extension might publish to Redis channel before gateway services are subscribe to that channel, causing the GameState to be out of sync.

Reply:
I think that in JS you should use an `async` function `launchGatewayService(someService)` to handle the startup of a service. In such way, you can do this:
```javascript
const allServices = [...];
const promises = allServices.forEach((s) => {return launchGatewayService(s)});
await Promise.all(promises);
```

I also saw you use `fork` to launch the gateway server. You may need a ping-pong message to know if your gateway server is ready.